### DISCUSSION TITLE
[DI] Is there a way to have synthetic default arguments?

### DISCUSSION THREAD

**Discussion by gaborjonas** [symfony/symfony#52980] [Q&A]
I'm adding the DI component to a legacy app, the piece of code looks something like this:
```php
$params = getFromDB();

$controller = new $controllerName($request, $params[0], $params[1]);
```

It's simple in PHP-DI
```php

$container->make($controllerName, [
   'param1' => 1,
   'param2' => 'string',
]);
```

I guess I'm looking for something like
```php
$container->services()
    ->defaults()
    ->bind('int $param1', ??)
    ->bind('string $param2', ??);
```

Is there a way to make it workgin with Symfony DI?
Thanks

**Comment by MatTheCat** [comment]
Have a look at configurators: https://symfony.com/doc/current/service_container/configurators.html
